test/e2e/framework/networking_utils.go

```diff
@@ -186,13 +186,12 @@ func (config *NetworkingTestConfig) DialFromContainer(protocol, containerIP, tar
 			if err := json.Unmarshal([]byte(stdout), &output); err != nil {
 				Logf("WARNING: Failed to unmarshal curl response. Cmd %v run in %v, output: %s, err: %v",
 					cmd, config.HostTestContainerPod.Name, stdout, err)
-				continue
-			}
-
-			for _, hostName := range output["responses"] {
-				trimmed := strings.TrimSpace(hostName)
-				if trimmed != "" {
-					eps.Insert(trimmed)
+			} else {
+				for _, hostName := range output["responses"] {
+					trimmed := strings.TrimSpace(hostName)
+					if trimmed != "" {
+						eps.Insert(trimmed)
+					}
```

Comment:
ah, yes I'm mixing them, but the idea was thay they'd have similar signatures and usage, so the case still applied. If I call it with an empty list, I want it to poll making sure the service/nodePort etc returns no results, not return with success if it encountered straight json.Unmarshal errors.